trace-dispatcher/src/Cardano/Logging/ConfigurationParser.hs

```diff
@@ -19,53 +20,108 @@ import qualified Data.ByteString.Char8 as BS
 import           Data.List (foldl')
 import qualified Data.Map.Strict as Map
 import           Data.Maybe (isJust)
-import           Data.Text (Text, split)
+import           Data.Text (Text, intercalate, split)
 import           Data.Yaml
 import           GHC.Generics
 
-
 import           Cardano.Logging.Types
 
-defaultConfig :: TraceConfig
-defaultConfig = emptyTraceConfig {
-  tcOptions = Map.fromList
-    [([] :: [Text],
-         [ ConfSeverity (SeverityF (Just Info))
-         , ConfDetail DNormal
-         , ConfBackend [Stdout HumanFormatColoured]
-         ])
-    ]
-  }
-
 -- -----------------------------------------------------------------------------
 -- Configuration file
 
+-- | The external representation of a configuration file
+data ConfigRepresentation = ConfigRepresentation {
+    traceOptions                  :: OptionsRepresentation
+  , traceOptionForwarder          :: Maybe TraceOptionForwarder
+  , traceOptionNodeName           :: Maybe Text
+  , traceOptionPeerFrequency      :: Maybe Int
+  , traceOptionResourceFrequency  :: Maybe Int
+  }
+  deriving (Eq, Ord, Show)
+
+instance AE.FromJSON ConfigRepresentation where
+    parseJSON (Object obj) = ConfigRepresentation
+                           <$> obj .: "TraceOptions"
+                           <*> obj .:? "TraceOptionForwarder"
+                           <*> obj .:? "TraceOptionNodeName"
+                           <*> obj .:? "TraceOptionPeerFrequency"
+                           <*> obj .:? "TraceOptionResourceFrequency"
+    parseJSON _ = mempty
+
+instance AE.ToJSON ConfigRepresentation where
+  toJSON ConfigRepresentation{..} = object
+    [ "TraceOptions"                  .= traceOptions
+    , "TraceOptionForwarder"          .= traceOptionForwarder
+    , "traceOptionNodeName"           .= traceOptionNodeName
+    , "TraceOptionPeerFrequency"      .= traceOptionPeerFrequency
+    , "traceOptionResourceFrequency"  .= traceOptionResourceFrequency
+    ]
```

Comment:
The capitalization of object keys doesn't align between `FromJSON` and `ToJSON`, is that intentional?
If there's a `Generic` instance for everything (not sure), one could used derived instances, or for capital initials something like:
```
instance FromJSON ConfigRepresentation where
    parseJSON = genericParseJSON defaultOptions {fieldLabeModifier =  <capitalize first letter>}
```

Reply:
Corrected the capitalisation. Thanks for pointing this out. It is only one unused Generic instance here, which I remove.